### PR TITLE
Change assert statements to raise statements

### PR DESCRIPTION
**Pull request by mikeessen**
Signed-off-by: Mike Essenmacher <essen@us.ibm.com>

### Description
    * This PR changes the assert statements to raise statements in the following parts:
        * onnx/helper.py
        * onnx/numpy_helper.py
        * onnx/reference/ops/op_resize.py
        * onnx/utils.py

### Motivation and Context
    * The assert statements could be optimized out when running with optimization.  
      Changing to raise keeps the validation.

### CODE REVIEW

File: onnx/helper.py [hunk 1]
```diff
@@ -563,7 +563,10 @@ def make_attribute(
         attr.type = AttributeProto.INT
     # string
     elif bytes_or_false is not False:
-        assert isinstance(bytes_or_false, bytes)
+        if not isinstance(bytes_or_false, bytes):
+            raise ValueError(
+                "{} must be an instance of {}.".format(bytes_or_false, bytes)
+            )
```
Comment (liqunfu) [inline onnx/helper.py:569]: Thanks for this PR. looks like this error message is "??? must be an instance of <class 'bytes'>.". Is it better to post "value must be an instance of <class 'bytes'>."

File: onnx/helper.py [hunk 2]
```diff
@@ -945,7 +948,8 @@ def str_list(str_elem: Callable[[_T], str], xs: Sequence[_T]) -> str:
 
 def printable_dim(dim: TensorShapeProto.Dimension) -> str:
     which = dim.WhichOneof("value")
-    assert which is not None
+    if which is None:
+        raise ValueError("{} cannot be {}.".format(which, None))
```
Comment (liqunfu) [inline onnx/helper.py:952]: 'None cannot be None.'?

**Review by liqunfu — CHANGES_REQUESTED** [review]
thanks for this PR! It looks good to me in general. My suggestion is to make error messages more understandable.

File: onnx/helper.py (hunk 1 above)
Comment (mikeessen) [reply onnx/helper.py]: Thanks for the comment, I updated the message.

File: onnx/helper.py (hunk 2 above)
Comment (mikeessen) [reply onnx/helper.py]: Thanks, I updated the message.

**Comment by mikeessen** [comment]
@iqunfu Thanks for the comments, I updated the messages.  Please take a look.

File: onnx/helper.py [hunk 3]
```diff
@@ -1097,7 +1102,10 @@ def strip_doc_string(proto: google.protobuf.message.Message) -> None:
     """
     Empties `doc_string` field on any nested protobuf messages
     """
-    assert isinstance(proto, google.protobuf.message.Message)
+    if not isinstance(proto, google.protobuf.message.Message):
+        raise ValueError(
+            "value must be an instance of {}.".format(google.protobuf.message.Message)
```
Comment (liqunfu) [inline onnx/helper.py:1107]: proto must be...

File: onnx/helper.py [hunk 4]
```diff
@@ -1078,7 +1082,8 @@ def printable_graph(graph: GraphProto, prefix: str = "") -> str:
     # body
     for node in graph.node:
         contents_subgraphs = printable_node(node, indent, subgraphs=True)
-        assert isinstance(contents_subgraphs[1], list)
+        if not isinstance(contents_subgraphs[1], list):
+            raise ValueError("value must be an instance of {}.".format(list))
```
Comment (liqunfu) [inline onnx/helper.py:1086]: contents_subgraphs[1] must be...

File: onnx/helper.py [hunk 5]
```diff
@@ -995,12 +997,14 @@ def printable_node(
...
-            assert isinstance(printed_attr_subgraphs[1], list)
+            if not isinstance(printed_attr_subgraphs[1], list):
+                raise ValueError("value must be an instance of {}.".format(list))
             graphs.extend(printed_attr_subgraphs[1])
             printed_attrs.append(printed_attr_subgraphs[0])
         else:
             printed = printable_attribute(attr)
-            assert isinstance(printed, str)
+            if not isinstance(printed, str):
+                raise ValueError("value must be an instance of {}.".format(str))
```
Comment (liqunfu) [inline onnx/helper.py:1007]: printed must be...

File: onnx/helper.py [hunk 6]
```diff
@@ -995,12 +997,14 @@ def printable_node(
     for attr in node.attribute:
         if subgraphs:
             printed_attr_subgraphs = printable_attribute(attr, subgraphs)
-            assert isinstance(printed_attr_subgraphs[1], list)
+            if not isinstance(printed_attr_subgraphs[1], list):
+                raise ValueError("value must be an instance of {}.".format(list))
```
Comment (liqunfu) [inline onnx/helper.py:1001]: printed_attr_subgraphs[1] must be...

File: onnx/helper.py (hunk 3 above)
Comment (mikeessen) [reply onnx/helper.py]: Updated, thank you.

File: onnx/helper.py (hunk 4 above)
Comment (mikeessen) [reply onnx/helper.py]: Updated, thank you.

File: onnx/helper.py (hunk 6 above)
Comment (mikeessen) [reply onnx/helper.py]: Updated, thank you.

File: onnx/helper.py (hunk 5 above)
Comment (mikeessen) [reply onnx/helper.py]: Updated, thank you.

**Review by liqunfu — APPROVED** [review]

**Review by liqunfu — APPROVED** [review]

File: onnx/helper.py [hunk 7]
```diff
@@ -359,8 +359,8 @@ def make_tensor(
     tensor.data_type = data_type
     tensor.name = name
 
-    if data_type == TensorProto.STRING:
-        assert not raw, "Can not use raw_data to store string type"
+    if data_type == TensorProto.STRING and raw:
+        raise ValueError("Can not use raw_data to store string type.")
```
Comment (justinchuby) [inline onnx/helper.py:363]: Should this be TypeError?

File: onnx/helper.py [hunk 8]
```diff
@@ -563,7 +563,8 @@ def make_attribute(
         attr.type = AttributeProto.INT
     # string
     elif bytes_or_false is not False:
-        assert isinstance(bytes_or_false, bytes)
+        if not isinstance(bytes_or_false, bytes):
+            raise ValueError(f"bytes_or_false must be an instance of {bytes}.")
```
Comment (justinchuby) [inline onnx/helper.py:567]: TypeError?

File: onnx/helper.py [hunk 9]
```diff
@@ -945,7 +946,8 @@ def str_list(str_elem: Callable[[_T], str], xs: Sequence[_T]) -> str:
 
 def printable_dim(dim: TensorShapeProto.Dimension) -> str:
     which = dim.WhichOneof("value")
-    assert which is not None
+    if which is None:
+        raise ValueError(f"which cannot be {None}.")
```
Comment (justinchuby) [inline onnx/helper.py:950]: TypeError?

File: onnx/helper.py [hunk 10]
```diff
@@ -1097,7 +1104,10 @@ def strip_doc_string(proto: google.protobuf.message.Message) -> None:
     """
     Empties `doc_string` field on any nested protobuf messages
     """
-    assert isinstance(proto, google.protobuf.message.Message)
+    if not isinstance(proto, google.protobuf.message.Message):
+        raise ValueError(
```
Comment (justinchuby) [inline onnx/helper.py:1108]: TypeError?

File: onnx/helper.py [hunk 11]
```diff
@@ -1078,7 +1084,8 @@ def printable_graph(graph: GraphProto, prefix: str = "") -> str:
     # body
     for node in graph.node:
         contents_subgraphs = printable_node(node, indent, subgraphs=True)
-        assert isinstance(contents_subgraphs[1], list)
+        if not isinstance(contents_subgraphs[1], list):
+            raise ValueError(f"contents_subgraphs[1] must be an instance of {list}.")
```
Comment (justinchuby) [inline onnx/helper.py:1088]: TypeError?

File: onnx/helper.py [hunk 12]
```diff
@@ -995,12 +997,16 @@ def printable_node(
...
+                raise ValueError(
+                    f"printed_attr_subgraphs[1] must be an instance of {list}."
+                )
             graphs.extend(printed_attr_subgraphs[1])
             printed_attrs.append(printed_attr_subgraphs[0])
         else:
             printed = printable_attribute(attr)
-            assert isinstance(printed, str)
+            if not isinstance(printed, str):
+                raise ValueError(f"printed must be an instance of {str}.")
```
Comment (justinchuby) [inline onnx/helper.py:1009]: TypeError?

File: onnx/helper.py [hunk 13]
```diff
@@ -995,12 +997,16 @@ def printable_node(
     for attr in node.attribute:
         if subgraphs:
             printed_attr_subgraphs = printable_attribute(attr, subgraphs)
-            assert isinstance(printed_attr_subgraphs[1], list)
+            if not isinstance(printed_attr_subgraphs[1], list):
+                raise ValueError(
+                    f"printed_attr_subgraphs[1] must be an instance of {list}."
```
Comment (justinchuby) [inline onnx/helper.py:1002]: TypeError?

File: onnx/helper.py (hunk 7 above)
Comment (mikeessen) [reply onnx/helper.py]: Yes, thanks.  Updated to TypeError, and also the other lines you suggested below.

**Review by justinchuby — APPROVED** [review]

**Comment by mikeessen** [comment]
@jcwchen Thanks for approving.  The PR has 8 workflows awaiting approval.  Please approve the workflows.

**Review by jcwchen — APPROVED** [review]

**Pull request merged**